spec/controllers/admin/videos_controller_spec.rb

```diff
@@ -0,0 +1,79 @@
+require 'spec_helper'
+
+describe Admin::VideosController do
+  it {is_expected.to use_before_action(:require_user)}
+  
+  describe "GET new" do
+    before {set_current_user}
+    it "sets @video to new video" do
+      set_current_admin
+      get :new
+      expect(assigns(:video)).to be_a_new Video
+    end
+    
+    it "sets flash error for regular user" do
+      get :new
+      expect(flash['danger']).to be_present
+    end
+    
+    it_behaves_like "require sign in" do
+      let(:action) {get :new}
+    end
+    
+    it_behaves_like "require admin" do
+      let(:action) {get :new}
+    end
+  end
+  
+  describe "POST create" do
+    before {set_current_user}
```

Comment:
Same issue here. In fact your later `set_current_admin` overrides this in both cases.